### PR TITLE
Fix crash due to txns in emit_log_hook_callback

### PR DESCRIPTION
**Pull request by nikkhils**
A weird crash was coming up while running the bgw_db_scheduler test
while using log_min_messages with DEBUG3 setting. So what's happening
here is that we register a RegisterXactCallback callback which gets
called at post-commit time. This remote_connection_xact_end callback
was issuing a debug3 level message. As part of its processing, the
emit_log_hook_callback function was trying to write this message
to a catalog table via a transaction. Obviously trying to start a
transaction at this post-commit stage will error out and leads to
weird behavior and crashes. In general it's not a good idea to have
emit_log_hook_callback functions start transactions because messages
can be emitted at any point of a transaction cycle, but we have to live
with the current code for now.
    
The fix is to reset the emit_log_hook in these xact_end callback
functions. Additionally the emit_log_hook_callback function is only
interested in elog/ereport messages with elevel LOG or above. So add
that check in the emit_log_hook_callback as well.
    
Fixes https://github.com/timescale/timescaledb-private/issues/899

### CODE REVIEW

**Comment by pmwkaa** [comment]
Is it possible to add a test case for it?

**Comment by nikkhils** [comment]
> Is it possible to add a test case for it?

This came up when I was running `bgw_db_scheduler` test with debug3 `log_min_messages`. So the test case is already there

**Review by mkindahl — APPROVED** [review]
LGTM

File: test/src/bgw/log.c [hunk 1]
```diff
@@ -107,6 +116,10 @@ emit_log_hook_callback(ErrorData *edata)
 	}
 	PG_CATCH();
 	{
+		/* If there was an error, rollback what was done before the error */
+		if (IsTransactionState())
+			AbortCurrentTransaction();
```
Comment (pmwkaa) [inline test/src/bgw/log.c:121]: Is that OK do it in the error handler here? Is it possible that `AbortCurrentTransaction()` can also throw an error?

**Review by pmwkaa — APPROVED** [review]

**Comment by svenklemm** [comment]
please squash commits before merging

File: test/src/bgw/log.c (hunk 1 above)
Comment (nikkhils) [reply test/src/bgw/log.c:121]: It's possible in which case a `RETHROW` might be required to handle it. But that's not needed here.

**Pull request merged**